src/gui/MixerLine.cpp

```diff
@@ -435,7 +435,7 @@ void MixerLine::setStrokeInnerInactive( const QColor & c )
 void MixerLine::selectColor()
 {
 	auto channel = Engine::mixer()->mixerChannel( m_channelIndex );
-	auto new_color = ColorChooser(this).withPalette(ColorChooser::Palette::Mixer)->getColor(channel->m_color);
+	auto new_color = ColorChooser(this).withPalette(ColorChooser::Palette::Mixer)->getColor(*channel->m_color);
```

Comment:
If `channel->m_color` doesn't have a value, the behavior is undefined. [`std::optional::value_or`](https://en.cppreference.com/w/cpp/utility/optional/value_or) might be useful, but it's hard to say what is the desired default value.

On the other hand(and out of scope), I found that using `ColorChooser` in this way looks weird: both `QColorDialog::setStandardColor`(called by `ColorChooser::withPalette`) and `QColorDialog::getColor` are static functions.

Reply:
Here's a quick color reference just for fun

[Default highlight of MixerLine](https://github.com/LMMS/lmms/blob/0b27497be2ac93551c4e243e43d7f0811ac0b889/src/gui/MixerLine.cpp#L183)

    backgroundActive();

[Background of a widget](https://doc.qt.io/qt-5/qpalette.html#ColorRole-enum)

    palette().color(QPalette::Window);

[Widget's `color` in the stylesheet](https://doc.qt.io/qt-6/stylesheet-reference.html#color-prop)

    palette().color(foregroundRole());

[LMMS green (depending on theme naturally)](https://github.com/LMMS/lmms/blob/6ae075edf3c052a923074b46ae1115718a621acd/data/themes/default/style.css#L1009)

    qApp->palette().color(QPalette::WindowText);

Reply:
I think the default mixer line colour (as defined in the stylesheet) would be an intuitive choice - that way, the dialog opens with the current mixer line colour, just as it would if a custom one were set.

Reply:
@DomClark's suggestion will look like this:
```suggestion
	auto new_color = ColorChooser(this).withPalette(ColorChooser::Palette::Mixer)->getColor(channel->m_color.value_or(backgroundActive()));
```

Reply:
Used DomClark suggestion based on PhySongs code example